.github/workflows/ci.yml

```
@@ -23,6 +23,23 @@ env:
   MAKEJOBS: '-j10'
 
 jobs:
+  test-each-commit:
+    name: 'test each commit'
+    runs-on: ubuntu-22.04
+    if: github.event_name == 'pull_request'
+    timeout-minutes: 360  # Use maximum time, see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes. Assuming a worst case time of 1 hour per commit, this leads to a --max-count=6 below.
+    env:
+      MAX_COUNT: '--max-count=6'
+    steps:
+      - uses: actions/checkout@v3
```

fafcd2e9ef1209d614de5763a2733098537919dd

Suggesting to rebase on top of the https://github.com/bitcoin/bitcoin/pull/28402 and
```suggestion
      - uses: actions/checkout@v4
```